partner_multi_relation/models/res_partner.py

```diff
@@ -171,3 +171,23 @@ def get_partner_type(self):
         """
         self.ensure_one()
         return 'c' if self.is_company else 'p'
+
+    @api.multi
+    def action_view_relations(self):
+        for contact in self:
+            relation_ids = self.env['res.partner.relation.all'].\
+                search(['|', ('this_partner_id', '=', contact.id),
+                       ('other_partner_id', '=', contact.id)])
+            action = self.env.\
+                ref('partner_multi_relation.action_res_partner_relation_all').\
+                read()[0]
+            action['context'] = {}
+            action['domain'] = [('id', 'in', relation_ids.ids)]
+            action['context'].\
```

Comment:
Merge the existing context (self.env.context, with the updated values in one assignment to action['context'].